### ISSUE TITLE
All (except 1) indexers failing tests with Ssl error:1000007d:SSL routines:OPENSSL_internal:CERTIFICATE_VERIFY_FAILED

### ISSUE THREAD

**Issue by aliekiddo** [Jackett/Jackett#13342]
### Have you checked our Troubleshooting page for your issue?

- [X] I have checked the Troubleshooting page

### Is there already an issue for your problem?

- [X] I have checked older issues, open and closed

### Have you read our Contributing Guidelines?

- [X] I have read the Contributing Guidelines

### Environment

```markdown
Info Environment version: 4.0.30319.42000 (/usr/local/lib/mono/4.5) 
2022-07-05 00:59:16.9713 Info OS version: Unix 13.1.6.0 (64bit OS) (64bit process) 
2022-07-05 00:59:16.9713 Info Jackett variant: Mono 
2022-07-05 00:59:16.9723 Info Running in Docker: No 
2022-07-05 00:59:16.9723 Info ThreadPool MaxThreads: 400 workerThreads, 200 completionPortThreads 
2022-07-05 00:59:16.9723 Info App config/log directory: /usr/local/jackett 
2022-07-05 00:59:16.9723 Info Using proxy: Disabled 
2022-07-05 00:59:16.9723 Info Using FlareSolverr: No 
2022-07-05 00:59:16.9723 Info Mono version: 6.8.0.123 (tarball Sat Jun 25 21:56:29 UTC 2022) 
2022-07-05 00:59:17.0005 Info Using HTTP Client: HttpWebClient
```

### Description

I can only successfully connect to one indexer (torrents.csv) everything else gives the ssl error:

> [*] (testing) Exception (*): Ssl error:1000007d:SSL routines:OPENSSL_internal:CERTIFICATE_VERIFY_FAILED  at /wrkdirs/usr/ports/lang/mono6.8/work/mono-6.8.0.123/external/boringssl/ssl/handshake_client.c:1132: The SSL connection could not be established, see inner exception.

I am running from a TrueNas jail, this is a brand new install because previously trying to update the plugin caused my old installation to not be able to connect to the webUI.

I've tried updating Jackett with the script described in the issue [here](https://github.com/Jackett/Jackett/issues/12432#issuecomment-946400786) but it didn't solve the issue for me and produced the errors in the attached screenshot.

### Logged Error Messages

```text
2022-07-04 14:44:35.4333 Error Error checking for updates.
System.Net.Http.HttpRequestException: The SSL connection could not be established, see inner exception. ---> System.Security.Authentication.AuthenticationException: Authentication failed, see inner exception. ---> Mono.Btls.MonoBtlsException: Ssl error:1000007d:SSL routines:OPENSSL_internal:CERTIFICATE_VERIFY_FAILED
  at /wrkdirs/usr/ports/lang/mono6.8/work/mono-6.8.0.123/external/boringssl/ssl/handshake_client.c:1132
  at Mono.Btls.MonoBtlsContext.ProcessHandshake () [0x00048] in <2deb8ccd8f0546038e28d46e7bcc1998>:0 
  at Mono.Net.Security.MobileAuthenticatedStream.ProcessHandshake (Mono.Net.Security.AsyncOperationStatus status, System.Boolean renegotiate) [0x000da] in <2deb8ccd8f0546038e28d46e7bcc1998>:0 
  at (wrapper remoting-invoke-with-check) Mono.Net.Security.MobileAuthenticatedStream.ProcessHandshake(Mono.Net.Security.AsyncOperationStatus,bool)
  at Mono.Net.Security.AsyncHandshakeRequest.Run (Mono.Net.Security.AsyncOperationStatus status) [0x00006] in <2deb8ccd8f0546038e28d46e7bcc1998>:0 
  at Mono.Net.Security.AsyncProtocolRequest.ProcessOperation (System.Threading.CancellationToken cancellationToken) [0x000fc] in <2deb8ccd8f0546038e28d46e7bcc1998>:0 
   --- End of inner exception stack trace ---
  at Mono.Net.Security.MobileAuthenticatedStream.ProcessAuthentication (System.Boolean runSynchronously, Mono.Net.Security.MonoSslAuthenticationOptions options, System.Threading.CancellationToken cancellationToken) [0x00262] in <2deb8ccd8f0546038e28d46e7bcc1998>:0 
  at System.Net.Http.ConnectHelper.EstablishSslConnectionAsyncCore (System.IO.Stream stream, System.Net.Security.SslClientAuthenticationOptions sslOptions, System.Threading.CancellationToken cancellationToken) [0x000c3] in <53402c1c880b4cf1bde830eb66c183f0>:0 
   --- End of inner exception stack trace ---
  at System.Net.Http.ConnectHelper.EstablishSslConnectionAsyncCore (System.IO.Stream stream, System.Net.Security.SslClientAuthenticationOptions sslOptions, System.Threading.CancellationToken cancellationToken) [0x00102] in <53402c1c880b4cf1bde830eb66c183f0>:0 
  at System.Net.Http.HttpConnectionPool.CreateConnectionAsync (System.Net.Http.HttpRequestMessage request, System.Threading.CancellationToken cancellationToken) [0x00322] in <53402c1c880b4cf1bde830eb66c183f0>:0 
  at System.Net.Http.HttpConnectionPool.WaitForCreatedConnectionAsync (System.Threading.Tasks.ValueTask`1[TResult] creationTask) [0x000a2] in <53402c1c880b4cf1bde830eb66c183f0>:0 
  at System.Net.Http.HttpConnectionPool.SendWithRetryAsync (System.Net.Http.HttpRequestMessage request, System.Boolean doRequestAuth, System.Threading.CancellationToken cancellationToken) [0x00089] in <53402c1c880b4cf1bde830eb66c183f0>:0 
  at System.Net.Http.DecompressionHandler.SendAsync (System.Net.Http.HttpRequestMessage request, System.Threading.CancellationToken cancellationToken) [0x000ca] in <53402c1c880b4cf1bde830eb66c183f0>:0 
  at FlareSolverrSharp.ClearanceHandler.SendAsync (System.Net.Http.HttpRequestMessage request, System.Threading.CancellationToken cancellationToken) [0x000d7] in <4f0872245bde4e2dabc08bde9c99ad32>:0 
  at System.Net.Http.HttpClient.FinishSendAsyncBuffered (System.Threading.Tasks.Task`1[TResult] sendTask, System.Net.Http.HttpRequestMessage request, System.Threading.CancellationTokenSource cts, System.Boolean disposeCts) [0x0017e] in <53402c1c880b4cf1bde830eb66c183f0>:0 
  at Jackett.Common.Utils.Clients.HttpWebClient.Run (Jackett.Common.Utils.Clients.WebRequest webRequest) [0x004ab] in <8c6f30ca1c444905915fe001acb7e52a>:0 
  at Jackett.Common.Utils.Clients.WebClient.GetResultAsync (Jackett.Common.Utils.Clients.WebRequest request) [0x001c2] in <8c6f30ca1c444905915fe001acb7e52a>:0
```

### Screenshots

![Screenshot_20220705-002715__01](https://user-images.githubusercontent.com/7974754/177225717-98a22d95-6a33-4fe3-8994-c75657ffa145.jpg)
![Screenshot_20220705-004025](https://user-images.githubusercontent.com/7974754/177225719-ee21d5ad-7ffe-4bc4-a3d1-5fda1174059f.jpg)

**Comment by ilike2burnthing** [comment]
Try:
https://github.com/Jackett/Jackett/issues/12341#issuecomment-957585781
and if that doesn't work:
https://github.com/fulder/iocage-plugin-jackett/issues/4#issuecomment-944931404

**Comment by aliekiddo** [comment]
I'm not sure if I'm doing it right but I couldn't find "DST Root CA X3" in the cert file in/ usr/local/share/certs/ca-root-nss.crt

I did find the file "2e5ac55d.0" in /etc/ssl/certs mentioned in the comment below and deleted it, but it is still not working.

I am trying to get my head around the info in the 2nd thread.

**Comment by aliekiddo** [comment]
I should also add that I've only noticed this problem within the last 2 weeks. I haven't downloaded very much recently but I am fairly certain it's been working since Sept 21 after the DST Root CA X3 cert expiry date.

**Comment by garfield69** [comment]
FYI the last time we made changes to the Jackett core was back on the 25th of Apri, relating to cors and dropping DPAPI and stuff.
All daily changes since then have been to individual indexers, and unlikely to be related to your issue.

**Comment by aliekiddo** [comment]
I've found and deleted the DSA Root CA X3 cert in the TrueNAS certs folder (as opposed to the Jackett jail folder) according to [this thread](https://www.truenas.com/community/threads/ssl-certificate-problem-certificate-has-expired-the-openssl-1-0-2-vs-letsencrypt-issue.95874/) but still nothing.

**Comment by CrazyMacGuy** [comment]
I seem to be running into this same issue, with a new install of TrueNAS Core 13.0 (TrueNAS-13.0-RELEASE) and Jackett manually updated to v0.20.1290 [using this method](https://github.com/jpbnetley/truenas-docs/blob/main/plugins/Jackett/jackett.md) since the version that gets installed from the community plugin list is v0.20.989.  This also seem to be impacting Sonarr that I just installed as well, but Radarr seems to be working fine.

**Comment by garfield69** [comment]
I do not know  much about your truenas community plugin version of Jackett, but once installed, can you not then use the Jackett dashboard and click on the  check-for-updates  button and be brought up to date that way?

**Comment by CrazyMacGuy** [comment]
> I do not know much about your truenas community plugin version of Jackett, but once installed, can you not then use the Jackett dashboard and click on the check-for-updates button and be brought up to date that way?

That usually works, yeah - but it was failing with the same sort of SSL error, so I had to use a manual method to see if the updated version would have the same issues (which it sadly did).

**Comment by ilike2burnthing** [comment]
@fulder any idea what the issue here might be?

**Comment by aliekiddo** [comment]
> I do not know much about your truenas community plugin version of Jackett, but once installed, can you not then use the Jackett dashboard and click on the check-for-updates button and be brought up to date that way?

Same, I've just tried that and it comes up against the same error, below is the log if that helps

I will try @CrazyMacGuy 's method with v0.20.1290

> 	
Error checking for updates.
System.Net.Http.HttpRequestException: The SSL connection could not be established, see inner exception. ---> System.Security.Authentication.AuthenticationException: Authentication failed, see inner exception. ---> Mono.Btls.MonoBtlsException: Ssl error:1000007d:SSL routines:OPENSSL_internal:CERTIFICATE_VERIFY_FAILED
  at /wrkdirs/usr/ports/lang/mono6.8/work/mono-6.8.0.123/external/boringssl/ssl/handshake_client.c:1132
  at Mono.Btls.MonoBtlsContext.ProcessHandshake () [0x00048] in <2deb8ccd8f0546038e28d46e7bcc1998>:0 
  at Mono.Net.Security.MobileAuthenticatedStream.ProcessHandshake (Mono.Net.Security.AsyncOperationStatus status, System.Boolean renegotiate) [0x000da] in <2deb8ccd8f0546038e28d46e7bcc1998>:0 
  at (wrapper remoting-invoke-with-check) Mono.Net.Security.MobileAuthenticatedStream.ProcessHandshake(Mono.Net.Security.AsyncOperationStatus,bool)
  at Mono.Net.Security.AsyncHandshakeRequest.Run (Mono.Net.Security.AsyncOperationStatus status) [0x00006] in <2deb8ccd8f0546038e28d46e7bcc1998>:0 
  at Mono.Net.Security.AsyncProtocolRequest.ProcessOperation (System.Threading.CancellationToken cancellationToken) [0x000fc] in <2deb8ccd8f0546038e28d46e7bcc1998>:0 
   --- End of inner exception stack trace ---
  at Mono.Net.Security.MobileAuthenticatedStream.ProcessAuthentication (System.Boolean runSynchronously, Mono.Net.Security.MonoSslAuthenticationOptions options, System.Threading.CancellationToken cancellationToken) [0x00262] in <2deb8ccd8f0546038e28d46e7bcc1998>:0 
  at System.Net.Http.ConnectHelper.EstablishSslConnectionAsyncCore (System.IO.Stream stream, System.Net.Security.SslClientAuthenticationOptions sslOptions, System.Threading.CancellationToken cancellationToken) [0x000c3] in <53402c1c880b4cf1bde830eb66c183f0>:0 
   --- End of inner exception stack trace ---
  at System.Net.Http.ConnectHelper.EstablishSslConnectionAsyncCore (System.IO.Stream stream, System.Net.Security.SslClientAuthenticationOptions sslOptions, System.Threading.CancellationToken cancellationToken) [0x00102] in <53402c1c880b4cf1bde830eb66c183f0>:0 
  at System.Net.Http.HttpConnectionPool.CreateConnectionAsync (System.Net.Http.HttpRequestMessage request, System.Threading.CancellationToken cancellationToken) [0x00322] in <53402c1c880b4cf1bde830eb66c183f0>:0 
  at System.Net.Http.HttpConnectionPool.WaitForCreatedConnectionAsync (System.Threading.Tasks.ValueTask`1[TResult] creationTask) [0x000a2] in <53402c1c880b4cf1bde830eb66c183f0>:0 
  at System.Net.Http.HttpConnectionPool.SendWithRetryAsync (System.Net.Http.HttpRequestMessage request, System.Boolean doRequestAuth, System.Threading.CancellationToken cancellationToken) [0x00089] in <53402c1c880b4cf1bde830eb66c183f0>:0 
  at System.Net.Http.DecompressionHandler.SendAsync (System.Net.Http.HttpRequestMessage request, System.Threading.CancellationToken cancellationToken) [0x000ca] in <53402c1c880b4cf1bde830eb66c183f0>:0 
  at FlareSolverrSharp.ClearanceHandler.SendAsync (System.Net.Http.HttpRequestMessage request, System.Threading.CancellationToken cancellationToken) [0x000d7] in <4f0872245bde4e2dabc08bde9c99ad32>:0 
  at System.Net.Http.HttpClient.FinishSendAsyncBuffered (System.Threading.Tasks.Task`1[TResult] sendTask, System.Net.Http.HttpRequestMessage request, System.Threading.CancellationTokenSource cts, System.Boolean disposeCts) [0x0017e] in <53402c1c880b4cf1bde830eb66c183f0>:0 
  at Jackett.Common.Utils.Clients.HttpWebClient.Run (Jackett.Common.Utils.Clients.WebRequest webRequest) [0x004ab] in <8c6f30ca1c444905915fe001acb7e52a>:0 
  at Jackett.Common.Utils.Clients.WebClient.GetResultAsync (Jackett.Common.Utils.Clients.WebRequest request) [0x001c2] in <8c6f30ca1c444905915fe001acb7e52a>:0 
  at Jackett.Common.Services.UpdateService.CheckForUpdates () [0x001a2] in <8c6f30ca1c444905915fe001acb7e52a>:0

**Comment by CrazyMacGuy** [comment]
Okay, so Sonarr and Jackett still appear to be using Mono, while Radarr is running on .NET Core 6.0.3. The only thing I haven't had time to try yet is rolling back the version of Mono that seems to have been successful for another user [on this thread](https://github.com/fulder/iocage-plugin-jackett/issues/4#issuecomment-945057287)

**Comment by aliekiddo** [comment]
> Okay, so Sonarr and Jackett still appear to be using Mono, while Radarr is running on .NET Core 6.0.3. The only thing I haven't had time to try yet is rolling back the version of Mono that seems to have been successful for another user [on this thread](https://github.com/fulder/iocage-plugin-jackett/issues/4#issuecomment-945057287)

I've given this a try too but it falls apart at 
# certsync
Command not found

(I think this should be cert-sync but this doesn't appear to be doing much either)

# pkg info -l mono | head -n 1
No package(s) matching mono

# pkg remove mono
Cannot get an advisory lock on a database, it is locked by another process

And still no change

**Comment by CrazyMacGuy** [comment]
I tried the process myself, and I had to be more specific with the pkg name instead of just "mono" since it didn't seem to be finding a match. Even after that, when I removed mono... it removed Jackett as well and I really didn't feel like tinkering anymore. So, I'm going to tinker some more and learn Docker! I'm migrating to TrueNAS Scale, which is Linux (Debian) based under the hood instead of FreeBSD.
